src/Model.php

```diff
@@ -806,24 +806,18 @@ public function get(string $field = null)
         if ($field === null) {
             // Collect list of eligible fields
             $data = [];
-            if ($this->only_fields) {
-                // collect data for actual fields
-                foreach ($this->only_fields as $field) {
-                    $data[$field] = $this->get($field);
-                }
-            } else {
-                // get all fields
-                foreach ($this->getFields() as $field => $f) {
-                    $data[$field] = $this->get($field);
-                }
+            foreach ($this->only_fields ?: array_keys($this->getFields()) as $field) {
+                $data[$field] = $this->get($field);
             }
 
             return $data;
         }
 
         $this->checkOnlyFieldsField($field);
 
-        if (array_key_exists($field, $this->data)) {
+        if ($field === $this->id_field) {
```

Comment:
I hit this several times - what is the purpose of `id` field instead of `data['id']`?

currently, they can very easily go out of sync as `id` is not magic property and `data` is public and used also multiple times in `Model` outside `get`/`set` functions